packages/core/src/storage/compare.ts

```diff
@@ -121,10 +132,11 @@ export class StorageLayoutComparator {
     updated: ParsedTypeDetailed,
     { allowAppend }: { allowAppend: boolean },
   ): TypeChange | undefined {
-    if (original.head !== updated.head) {
+    if (updated.head.startsWith('t_function')) {
+      return this.getVisibilityChange(original, updated);
+    } else if (original.head !== updated.head) {
       return { kind: 'obvious mismatch', original, updated };
     }
```

Comment:
I prefer this order:

```suggestion
    if (original.head !== updated.head) {
      return { kind: 'obvious mismatch', original, updated };
    }

    if (updated.head.startsWith('t_function')) {
      return this.getVisibilityChange(original, updated);
    }

```

Reply:
I don't think the above will work. That's why I had them ordered the other way. Because the head will never match if visibility is changed. So the second if statement will never trigger. That being said the "obvious mismatch" does work in that situation. The real error was caused by the fact that the `t_function` types weren't being caught in the switch statement.

Reply:
Ahh yes you're right, my bad.

Just removing else then:

```suggestion
    if (updated.head.startsWith('t_function')) {
      return this.getVisibilityChange(original, updated);
    }

    if (original.head !== updated.head) {
      return { kind: 'obvious mismatch', original, updated };
    }

```